libkmod/libkmod-internal.h

```diff
@@ -10,22 +10,14 @@
 
 #include "libkmod.h"
 
-static _always_inline_ _printf_format_(2, 3) void kmod_log_null(const struct kmod_ctx *ctx,
-								const char *format, ...)
-{
-}
-
 #define kmod_log_cond(ctx, prio, arg...)                                          \
 	do {                                                                      \
-		if (kmod_get_log_priority(ctx) >= prio)                           \
+		if ((ENABLE_DEBUG || (!ENABLE_DEBUG && prio != LOG_DEBUG)) &&     \
+		    kmod_get_log_priority(ctx) >= prio)                           \
 			kmod_log(ctx, prio, __FILE__, __LINE__, __func__, ##arg); \
 	} while (0)
 
-#ifdef ENABLE_DEBUG
```

Comment:
only because you mandate logging... which means all the (sometimes very verbose strings) will be present in the binary